File: 프로그래머스/Lv. 2/두 원 사이의 정수 쌍.py
import math

def solution(r1, r2):
    count = 0
    
    for x in range(0, r2 + 1):
        max_y = math.floor(math.sqrt(r2 ** 2 - x ** 2))
        min_y = math.ceil(math.sqrt(r1 ** 2 - x ** 2)) if x**2 < r1**2 else 0
        
        count += (max_y - min_y + 1)
        
    return (count - (r2 - r1 + 1)) * 4

File: 프로그래머스/Lv. 2/두 원 사이의 정수 쌍.java
class Solution {
    public long solution(int r1, int r2) {
        long answer = 0;
        
        for (int x = 0; x < r2 + 1; x++) {
            int maxY = (int)Math.floor(Math.sqrt(Math.pow(r2, 2) - Math.pow(x, 2)));
            int minY;
            if (x < r1) {
                minY = (int)Math.ceil(Math.sqrt(Math.pow(r1, 2) - Math.pow(x, 2)));
            } else {
                minY = 0;
            }
            
            answer += (maxY - minY + 1);
        }
        
        return (answer - (r2 - r1 + 1)) * 4;
    }
}
